### PR TITLE
agent: return the default ACL policy to callers as a header

### PR DESCRIPTION
**Pull request by rboyer**
Header is: X-Consul-Default-ACL-Policy=<allow|deny>

This is of particular utility when fetching matching intentions, as the
fallthrough for a request that doesn't match any intentions is to
enforce using the default acl policy.

### CODE REVIEW

**Review by picatz — APPROVED** [review]

File: api/api.go [hunk 1]
```diff
@@ -254,6 +254,11 @@ type QueryMeta struct {
 	// CacheAge is set if request was ?cached and indicates how stale the cached
 	// response is.
 	CacheAge time.Duration
+
+	// DefaultACLPolicy is used to control the ACL interaction when there is no
+	// defined policy. This can be "allow" which means ACLs are used to
+	// black-list, or "deny" which means ACLs are white-lists.
```
Comment (freddygv) [inline api/api.go:260]:
```suggestion
	// deny-list, or "deny" which means ACLs are allow-lists.
```

**Review by freddygv — COMMENTED** [review]
I think it would be useful to add a test to [http_test.go](https://github.com/hashicorp/consul/blob/master/agent/http_test.go#L375) that verifies the header is being set from the agent config.

File: api/api.go (hunk 1 above)
Comment (rboyer) [reply api/api.go:260]: I actually lifted this comment pretty much as-is from `agent/config/runtime.go` and `agent/consul/config.go` so i'll fix them all.

**Comment by rboyer** [comment]
@freddygv I've added tests, a changelog entry, and updated the website.

**Review by freddygv — APPROVED** [review]
:shipit:

**Pull request merged**